packages/sdk/src/sdkBase.ts

```diff
@@ -1063,42 +1062,53 @@ export class NxtpSdkBase {
               throw new Error("No taskId returned");
             }
             this.logger.info("Submitted using Gelato Relayer", requestContext, methodContext, { data });
-            gelatoSuccess = true;
-            break;
+
+            try {
+              const response = await fulfillTxProm;
+              const ret = {
+                transactionHash: response.transactionHash,
+                chainId: response.txData.receivingChainId,
+              };
+              this.logger.info("Method complete", requestContext, methodContext, ret);
+              return { transactionResponse: ret };
+            } catch (e) {
+              throw e.message.includes("Evt timeout")
+                ? new FulfillTimeout(txData.transactionId, FULFILL_TIMEOUT, params.txData.receivingChainId, {
+                    requestContext,
+                    methodContext,
+                  })
+                : e;
+            }
           } catch (err) {
             this.logger.error("Error using Gelato Relayer", requestContext, methodContext, jsonifyError(err), {
-              attemptNum: ii + 1,
+              attemptNum: i + 1,
```

Comment:
do you want to have any delay logic here?

Reply:
Did I take that out? woops, should be re-added